Taskfile.yaml

```diff
@@ -82,10 +82,15 @@ tasks:
   order_service:build_docker: docker build --target order-service -t shopping-cart/order-service .
   ########## Cleanup ###################################################################################################
   clean:protobuf: find api/proto -type f -name "*.pb.go" -delete
+  clean:containers: CONTAINERS="$(docker ps --format \'\{\{.Names\}\}\' | grep 'shoppingcart')"; CONTAINERS=${CONTAINERS//\'}; echo -e "Containers found :\n${CONTAINERS}"; if [[ -z "$CONTAINERS" ]]; then echo "No shopping cart containers were found"; else docker rm ${CONTAINERS}; fi
+  clean:images: IMGS="$(docker images --format \'\{\{.Tag\}\}\' | grep 'shoppingcart')"; IMGS=${IMGS//\'}; echo -e "Images found :\n${IMGS}"; if [[ -z "$IMGS" ]]; then echo "No shopping cart images were found"; else docker rmi -f $IMGS; fi
   clean:
     - task: clean:protobuf
+    - task: clean:containers
+    - task: clean:images
```

Comment:
The `clean` commands are used for cleaning up the working directory of the project and shouldn't meddle with the state of the host services like Docker. Cleaning up the Docker's daemon can be done globally and periodically with the `docker system prune` command (in our case probably at the end of the quarter).

What's more, calling `task clean` when the Docker daemon is not running will make the whole command fail.